lib/common/io.c

```diff
@@ -33,35 +33,56 @@
 #include <crm/common/util.h>
 
 /*!
+ * \internal
  * \brief Create a directory, including any parent directories needed
  *
  * \param[in] path_c Pathname of the directory to create
  * \param[in] mode Permissions to be used (with current umask) when creating
  *
- * \note This logs errors but does not return them to the caller.
+ * \return pcmk_ok on success, -errno otherwise
```

Comment:
Would it be worth converting this function to returning a standard pacemaker return code now too?

Reply:
Whoops, that had been sitting around as a patch for a long time. Will do.